R/lint.R

```diff
@@ -88,9 +88,11 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
   }
 
   lints <- list()
-  for (expr in source_expressions$expressions) {
-    for (linter in names(linters)) {
-      lints[[length(lints) + 1L]] <- get_lints(expr, linter, linters[[linter]], lint_cache, source_expressions$lines)
+  if (!is_tainted(source_expressions$lines)) {
```

Comment:
does that mean such files go undetected?

what if I have an encoding issue without realizing it, and am trying to figure out why `a = 1` isn't triggering assignment linter, how do I diagnose that?

Reply:
partially orthogonal: should we make a note of this behavior in the ?lint documentation?

Reply:
It will produce an `error` lint. Only other lints are disabled and only if the parse error leaves butchered `file_lines`.

Reply:
> partially orthogonal: should we make a note of this behavior in the ?lint documentation?

Good idea. "If files contain unparseable encoding problems, only the encoding problem will be linted to avoid unintelligible error messages from other linters"?

Reply:
SGTM. LGTM otherwise